bolt-simple-taproot.md

```diff
@@ -0,0 +1,994 @@
+be achieved by computing the output key point as `Q = P +
+int(hashTapTweak(bytes(P)))G`". We refer to BIP 86 for the specifics of this
+derivation.
+
+By using the musig2 sorting routine, we inherit the lexicographical key
+ordering from the base segwit channels. Throughout the lifetime of a channel
+any keys aggregated via musig2 also are assumed to use the `KeySort` routine
+
+### Commitment Transactions
+
+#### To Local Outputs
+
+For the simple taproot commitment type, we use the same flow of revocation or
+delay, but instead left the revocation case into the taproot key-spend case.
+
+The new output has the following form:
+
+  * `OP_1 to_local_output_key`
+  * where:
+    * `to_local_output_key = revocationpubkey + tagged_hash("TapTweak", revocationpubkey || to_delay_script_root)`
+    * `to_delay_script_root = tapscript_root([to_delay_script])`
+    * `to_delay_script` is the delay script:
+        ```
+        <local_delayedpubkey> OP_CHECKSIG
+        <to_self_delay> OP_CHECKSEQUENCEVERIFY OP_DROP
+        ```
+
+The parity (even or odd) of the y-coordinate of the derived
+`to_local_output_key` MUST be known in order to derive a valid control block.
+
+The `tapscript_root` routine constructs a valid taproot commitment according to
+BIP 341+342. Namely, a leaf version of `0xc0` MUST be used. As there's only a
+single script, one can derive the tapscript root as:
+```
+tapscript_root = tagged_hash("TapLeaf", 0xc0 || compact_size_of(to_delay_script) || to_delay_script)
+```
+
+In the case of a commitment breach, the `to_delay_script_root` can be used
+along side `<revocationpubkey>` to derive the private key needed to sweep the
+top-level key spend path. A valid witness is then just:
+```
+<revocation_sig>
+```
+
+In the case of a routine force close, the script path must be revealed so the
+broadcaster can sweep their funds after a delay. The control block to spend is
+only `33` bytes, as it just includes the internal key (along with the y-parity
+bit and leaf version):
+```
+delay_control_block = (output_key_y_parity | 0xc0) || revocationpubkey
+```
+
+A valid witness is then:
+```
+<local_delayedsig> <to_delay_script> <delay_control_block>
+```
+
+As with base channels, the `nSequence` field must be set to `to_self_delay`.
+
+#### To Remote Outputs
+
+As we inherit the anchor output semantics we want to ensure that the remote
+party can unilaterally sweep their funds after the 1 block CSV delay. In order
+to achieve this property, we'll re-use the `_funding_key` here: its in
+the best interest of the other party to enforce these semantics (mempool
+pinning mitigation), so the remote party will be forced to always take the
+script reveeal path.
+
+The to remote output has the following form:
+
+  * `OP_1 to_remote_output_key`
+  * where:
+    * `to_remote_output_key = combined_funding_key + tagged_hash("TapTweak", combined_funding_key || to_remote_script_root)`
```

Comment:
Additionally, my understanding of Taproot tweak calculations is that the preimage first contains the Merkle root, and then the internal key, so it would actually be
`tweak = tagged_hash("TapTweak", to_remote_script_root || combined_funding_key)`